### PR TITLE
Release 1.1.7: More robust reboot check and update docs

### PR DESCRIPTION
**Pull request by kreeuwijk**
* Adds linting corrections
* Adds missing inline documentation back, after changing to 100% auto-generated REFERENCE.md
* Changes the reboot checks to verify based on script exitcode, not script output

### CODE REVIEW

**Comment by kreeuwijk** [comment]
Fixes #53, #67 and #71

File: metadata.json [hunk 1]
```diff
@@ -102,7 +102,7 @@
       "version_requirement": ">= 6.0.0 < 8.0.0"
     }
   ],
-  "pdk-version": "2.3.0",
-  "template-url": "pdk-default#2.3.0",
-  "template-ref": "tags/2.3.0-0-g8aaceff"
+  "pdk-version": "2.5.0",
```
Comment (bastelfreak) [inline metadata.json:105]: There is really a lot going on in this PR. Could you submit the pdk update in a dedicated PR? that makes it easier to review and gives a better changelog. The current PR title doesn't really reflect all the changes.

Comment (kreeuwijk) [reply metadata.json:105]: Fresh PR [here](https://github.com/puppetlabs/puppetlabs-patching_as_code/pull/81), I can rebase this PR after that one has been merged

**Review by psreed — APPROVED** [review]

File: metadata.json (hunk 1 above)
Comment (binford2k) [reply metadata.json:105]: could we get some of the commits squashed too?
